plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergFileWriterFactory.java

```diff
@@ -261,4 +275,23 @@ private IcebergFileWriter createOrcWriter(
             throw new TrinoException(ICEBERG_WRITER_OPEN_ERROR, "Error creating ORC file", e);
         }
     }
+
+    public static OrcWriterOptions getOrcWriterOptions(Map<String, String> storageProperties, OrcWriterOptions orcWriterOptions)
+    {
+        if (storageProperties.containsKey(ORC_BLOOM_FILTER_COLUMNS_KEY)) {
+            if (!storageProperties.containsKey(ORC_BLOOM_FILTER_FPP_KEY)) {
+                throw new TrinoException(HIVE_INVALID_METADATA, format("FPP for bloom filter is missing"));
+            }
+            try {
+                double fpp = parseDouble(storageProperties.get(ORC_BLOOM_FILTER_FPP_KEY));
+                return orcWriterOptions
+                        .withBloomFilterColumns(ImmutableSet.copyOf(COLUMN_NAMES_SPLITTER.splitToList(storageProperties.get(ORC_BLOOM_FILTER_COLUMNS_KEY))))
+                        .withBloomFilterFpp(fpp);
```

Comment:
Nit: This saves a little bit of converting from OrcWriterOptions and the Builder
```suggestion
                return OrcWriterOptions.builderFrom(orcWriterOptions)
                        .setBloomFilterColumns(ImmutableSet.copyOf(COLUMN_NAMES_SPLITTER.splitToList(storageProperties.get(ORC_BLOOM_FILTER_COLUMNS_KEY))))
                        .setBloomFilterFpp(fpp)
                        .build();
```

Reply:
Done.